util/version/version.go

```diff
@@ -173,22 +173,6 @@ func (v buildIdentifier) compare(o buildIdentifier) int {
 	}
 }
 
-// CompareWithBuildIdentifiers compares two versions a and b.
-// Perfoms a standard version compare between a and b. If the versions
-// are equal, build identifiers will be used to compare further.
-//   -1 == a is less than b.
-//   0 == a is equal to b.
-//   1 == a is greater than b.
-// Deprecated: Use Compare(a, b, WithBuildTags()) instead.
-func CompareWithBuildIdentifiers(a semver.Version, b semver.Version) int {
-	if comp := a.Compare(b); comp != 0 {
-		return comp
-	}
-	biA := newBuildIdentifiers(a.Build)
-	biB := newBuildIdentifiers(b.Build)
-	return biA.compare(biB)
-}
-
```

Comment:
`CompareWithBuildIdentifiers` is still being called in `Compare` https://github.com/kubernetes-sigs/cluster-api/blob/cca725c1ca50ee58ca1d4531dbeb2bb7a92669c7/util/version/version.go#L232

The functionality should be moved to `Compare` or `CompareWithBuildIdentifiers` should be made unexportable by renaming to `compareWithBuildIdentifiers` and using this internal implementation. Not a big fan of making unexportable since it can potentially create confusion. The maintainers could give more insights into the implementation that would be desired.

Similarly, tests for `CompareWithBuildIdentifiers` need to be removed and similar behaviour should be tested in tests for `Compare`.

Reply:
I think it would be good to inline `CompareWithBuildIdentifiers` in l.216

Regarding tests, I think we can ~ move the testcases from TestCompareWithBuildIdentifiers to TestCompare.
But please replace those funcs:
```go
			b: func() semver.Version {
				v, _ := semver.ParseTolerant("v1.20.1+xyz.a")
				return v
			}(),
```
through
```go
			bVersion: semver.MustParse("v1.20.1+xyz.a")
```
(and of course `options:  []CompareOption{WithBuildTags()},` has to be used then)

Reply:
@palnabarun @sbueringer Thank you for reviewing the PR. 
When I moved the test cases from TestCompareWithBuildIdentifiers to TestCompare and used the func MustParse, I see the following error: 
`panic: semver: Parse(v1.20.1): Invalid character(s) found in major number "v1"`
Is it okay to change the values from `v1.20.1+xyz.a` to `1.20.1+xyz.a` ?

Reply:
My bad, missed the tolerant part.
Can you introduce this func and use it for the cases which were using ParseTolerant before?
```go
func mustParseTolerant(s string) semver.Version {
	v, err := semver.ParseTolerant(s)
	if err != nil {
		panic(`semver: ParseTolerant(` + s + `): ` + err.Error())
	}
	return v
}
```